extensions/arc/src/ui/dialogs/connectControllerDialog.ts

```diff
@@ -88,8 +88,21 @@ export class ConnectToControllerDialog {
 		if (!this.urlInputBox.value || !this.usernameInputBox.value || !this.passwordInputBox.value) {
 			return false;
 		}
+		let url = this.urlInputBox.value;
```

Comment:
tests 😄

Reply:
This code seems interesting enough to maybe warrant it as a utility class to construct URLs from user input.

Reply:
Currently I doubt we'd ever need to use it outside this one particular place in this extension. I could potentially see this being a useful thing across multiple extensions but I'm not sure the use case is big enough to warrant the overhead of having to call through ext host/ext API. Do you see it being valuable elsewhere?

Reply:
What I meant was if it's a class it'd be easier to copy paste it to be used elsewhere if the need arises! But please don't block the PR on that :)